assets/js/blocks/reviews/reviews-by-product/edit.tsx

```diff
@@ -25,19 +25,18 @@ import {
 	getSharedReviewListControls,
 } from '../edit-utils.js';
 
-/**
- * Component to handle edit mode of "Reviews by Product".
- *
- * @param {Object}            props                Incoming props for the component.
- * @param {Object}            props.attributes     Incoming block attributes.
- * @param {function(any):any} props.debouncedSpeak
- * @param {function(any):any} props.setAttributes  Setter for block attributes.
- */
+interface ReviewsByProductEditorProps {
+	name: string;
+	attributes: BlockAttributes;
+	setAttributes: ( attrs: BlockAttributes ) => void;
+	debouncedSpeak: ( message: string ) => void;
+}
+
```

Comment:
Hey @albarin, Thanks for adding types to this file. Great job! 🙌🏻

What do you think about extending `BlockEditProps` like this:

```tsx
import type { BlockAttributes, BlockEditProps } from '@wordpress/blocks';
export interface ReviewsByProductEditorProps
	extends BlockEditProps< BlockAttributes > {
	debouncedSpeak: ( label: string ) => void;
}
```

Please let me know your thoughts on this!

By the way, it might be worth considering creating a new type specifically for the attributes instead of using the generic type BlockAttributes. Currently, BlockAttributes is defined as:
```tsx
export type BlockAttributes = Record<string, any>;
```

As you can see, it is a very generic type.